### PR TITLE
Add ECVRF to move api

### PR DESCRIPTION
**Pull request by jonas-lj**
Closing #7038

### CODE REVIEW

File: crates/sui-framework/sources/crypto/ecvrf.move [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    /// @param proof: The proof of validity of the output.
+    /// A native move wrapper around the Ristretto ECVRF. Returns true if the proof is valid and corresponds to the given output.
+    native fun native_ecvrf_verify(hash: &vector<u8>, alpha_string: &vector<u8>, public_key: &vector<u8>, proof: &vector<u8>): bool;
+
+    /// @param hash: The hash/output from a ECVRF to be verified.
+    /// @param alpha_string: Input/seed to the ECVRF used to generate the output.
+    /// @param public_key: The public key corresponding to the private key used to generate the output.
+    /// @param proof: The proof of validity of the output.
+    /// Verify a proof for a Ristretto ECVRF. Returns true if the proof is valid and corresponds to the given output.
+    public fun ecvrf_verify(hash: &vector<u8>, alpha_string: &vector<u8>, public_key: &vector<u8>, proof: &vector<u8>): bool {
```
Comment (joyqvq) [inline crates/sui-framework/sources/crypto/ecvrf.move:18]: why not make this `public native fun`?

File: crates/sui-framework/tests/crypto/ecvrf_tests.move [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+    #[test]
+    fun test_ecvrf_verify() {
+        // Test vector produced with Fastcrypto
+        let output = x"4ad05aafdaf4f0c76ac1ec46507431de2d81876ff95dd097334d2c257492afe41fe21900ef04e4e1f377143e24e38cf104fc5f983d28f940d4c0721d0823a3af";
+        let alpha_string = b"Hello, world!";
+        let public_key = x"2cf96313347d5f1f347464d1f59889b5c736a17e19bf899827aa3400733ea44d";
+        let proof = x"445b02b99b1484e22325b0ad609f666df3d7099c90ed82a15d731ec42603fa4279ef4781ec2b88355dc08f5cbdf6cdae511836d5b5a5566a12fc56db4d8c3f9bf1c85186719706282c525df4b0477502";
+        assert!(ecvrf::ecvrf_verify(&output, &alpha_string, &public_key, &proof), 0);
+    }
```
Comment (joyqvq) [inline crates/sui-framework/tests/crypto/ecvrf_tests.move:16]: add some invalid tests

File: crates/sui-framework/sources/crypto/ecvrf.spec.move [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+spec sui::ecvrf {
+    spec ecvrf_verify {
+        // TODO: temporary mockup.
```
Comment (joyqvq) [inline crates/sui-framework/sources/crypto/ecvrf.spec.move:6]: what is this for?

Comment (jonas-lj) [reply crates/sui-framework/sources/crypto/ecvrf.spec.move:6]: I'm not sure. It seems like the other crypto modules have it. I'll remove it for now.

**Review by joyqvq — APPROVED** [review]

**Review by kchalkias — COMMENTED** [review]
Ideally we'd need an example smart contract + tool to educate users re how to play with ECVRF

**Pull request merged**